pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2515,8 +2526,8 @@ public void readEntryComplete(Entry entry, Object ctx) {
 
     protected Response internalPeekNthMessage(String subName, int messagePosition, boolean authoritative) {
         // If the topic name is a partition name, no need to get partition topic metadata again
-        if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName,
-                authoritative, false).partitions > 0) {
+        if (!topicName.isPartitioned() && getPartitionedTopicMetadataAsync(topicName,
```

Comment:
It seems that changing here will not make any difference. Since https://github.com/apache/pulsar/pull/14039 is fixing the call sync method here. I think it's better not to change here.

Reply:
yes, right, this is a mistake for modifying here.